### PR TITLE
fix: correct router logic

### PR DESCRIPTION
**Pull request by Himself65**
Now, we only listen to the `routeChangeStart` and don't case other things.

### CODE REVIEW

File: apps/web/src/hooks/__tests__/use-recent-views.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
+/**
```
Comment (himself65) [inline apps/web/src/hooks/__tests__/use-recent-views.spec.tsx:1]: I moved this test into a new file because I found that next-router-mock has some corner case which might not isolate with other test case

File: packages/workspace/src/atom.ts [hunk 2]
```diff
@@ -32,12 +33,49 @@ export const rootCurrentWorkspaceIdAtom = atomWithStorage<string | null>(
...
 export const rootCurrentPageIdAtom = atomWithStorage<string | null>(
   'root-current-page-id',
   null,
   createJSONStorage(() => sessionStorage)
 );
 
+rootCurrentPageIdAtom.onMount = set => {
+  if (typeof window !== 'undefined') {
+    const callback = (url: string) => {
+      const value = url.split('/')[3];
```
Comment (himself65) [inline packages/workspace/src/atom.ts:64]: hard code here, but it's ok because we don't have other use cases

File: apps/web/src/layouts/workspace-layout.tsx [hunk 3]
```diff
@@ -137,13 +134,21 @@ export const AllWorkspaceContext = ({
...
-  useSyncRouterWithCurrentPageId(router);
-  useRouterWithWorkspaceIdDefense(router);
   const metadata = useAtomValue(rootWorkspacesMetadataAtom);
   const exist = metadata.find(m => m.id === workspaceId);
+  const router = useRouter();
+  const push = router.push;
+  useEffect(() => {
+    const id = setTimeout(() => {
+      if (!exist) {
+        void push('/');
```
Comment (himself65) [inline apps/web/src/layouts/workspace-layout.tsx:145]: This avoids entering a nonexisting workspace. we will fallback to the index page

File: apps/web/src/providers/modal-provider.tsx [hunk 4]
```diff
@@ -152,7 +152,7 @@ export function Modals() {
               setOpenCreateWorkspaceModal(false);
               setOpenWorkspacesModal(false);
               setCurrentWorkspace(id);
-              return jumpToSubPath(id, WorkspaceSubPath.SETTING);
+              return jumpToSubPath(id, WorkspaceSubPath.ALL);
```
Comment (pengx17) [inline apps/web/src/providers/modal-provider.tsx:155]: I think this is a requirement for creating new workspace that we will redirect to the settings pages after creation.
